tests/integration/telemetry/stackdriver/testdata/server_access_log.json.tmpl

```diff
@@ -26,7 +26,7 @@
         "source_version": "v1",
         "source_workload": "clt-{{ .ClusterName }}-v1",
         "protocol": "grpc",
-        "log_sampled": "true",
+        "log_sampled": "false",
```

Comment:
this is from removal of "access log policy"

Reply:
Does ASM use this setting? We need some coverage - but not sure if this is tested here or elsewhere.

Reply:
There's no docs, and it's impossible to use on MCP. I don't think there's any usage based on ~15min of digging

Reply:
Seems like a half baked featured that was never completed from what I can tell

Reply:
@lei-tang can you confirm? this has a high potential of a surprise bill if it's used and we miss it.

Reply:
even if it is somehow configured, I am fairly sure it bills exactly the same. It's purely an opaque label .

Which doesn't make much sense, so either I am missing something or the feature was never really completed

Reply:
The bill for access logs is substantial. Let's make sure that "Telemetry API" always configures "ERRORS_ONLY" on outbound which requires the sampler I think.

Reply:
I'll double check what the mode is but just to clarify, this PR only changes behavior if someone is configured an undocumented and unsupported config to set access_logging_policy (not possible in MCP, either)

For anyone else there is no change at all. 

even if they did set it, though, it has nothing to do with billing as far as I can tell. It does *not* sample despite the name it only sets a bool flag that has no impact

Reply:
"Set an access log policy" with incluster control plane is documented at: https://cloud.google.com/service-mesh/docs/observability/accessing-logs#set_an_access_log_policy.

If this feature is removed, documents need to be updated.

Reply:
Thanks I did not find that during my search (largely because it never actually says *how* to configure it, so I assume no one actually knows).

Even with this, the feature does not work and isn't usable with MCP, so I cannot imagine anyone is using it. I would still encourage us to move forward with this PR as-is and I will take a follow-up to update the docs.

If there is a disagreement I can add it back, but we are maintaining a ton of code that does nothing

Reply:
If this feature is not used by customers, it sounds good to me to remove this feature. If it has usage, I think a feature deprecation procedure may need to be followed.
 
Xinnan (@richardwxn) is the expert on feature usage data. Xinnan, do you know the usage data for this feature?

Reply:
> Even with this, the feature does not work and isn't usable with MCP, so I cannot imagine anyone is using it. I would still encourage us to move forward with this PR as-is and I will take a follow-up to update the docs.

I vaguely remember a user explicitly requested it and is well aware how this works. We have to check the usage first.

Reply:
Thanks for pushing on this - I think I was wrong.

I went through the docs, and they were incorrect initially, and now provide no information on how to configure it. Here is some options, in my order of preference:

1. Remove everything, as this PR does. Users can manually apply the EnvoyFilter if they need this.
2. Remove everything, as this PR does, but add a pilot env var to set this.
3. Keep just access_log_policy EnvoyFilter

In all cases I will also go back and add the tests back for this feature.

Reply:
Access log policy only works with stackdriver, so it can also be tied to stackdriver provider API. But I'm fine with any installation time method or leave EnvoyFilter for it as-is.

Reply:
Done, I did option (1) and added the test back with this enabled.

Reply:
As with Option 1, user manually applies the EnvoyFilter. I think a user guide is needed, including the content of the EnvoyFilter for access_log_policy.

Reply:
The user guide today already says "go figure it out". It doesn't say _how_ to do it at all. So I can improve the doc, but to be clear its already ~broken.